imi/src/main/java/org/motechproject/nms/imi/service/impl/CdrFileServiceImpl.java

```diff
@@ -90,6 +167,8 @@ private void parseDetailRecord(CallDetailRecordDto cdr, Map<String, CallSummaryR
                 record.setSecondsPlayed(cdr.getMsgPlayDuration());
             }
 
+            //todo: maybe verify the attemptCount provided by IMI and see if there are holes?
+
             /*
                Update the final status
 
```

Comment:
This comment is about line 178.

Can we accurately calculate the final status if the records aren't sorted?  If we get a success then we know that is the final status.  But all failure codes could be the actual final status and just because one comes later in a non-sorted file doesn't make it the final status.

There isn't anything we can do, just something we should note.